templates/k0s.yaml

```diff
@@ -40,7 +40,15 @@ spec:
     adminPort: 8133
     agentPort: 8132
   network:
-    calico: null
+    calico:
+      mode: vxlan
+      vxlanPort: 4789
+      vxlanVNI: 4096
+      mtu: 1450
+      wireguard: false
+      flexVolumeDriverPath: /usr/libexec/k0s/kubelet-plugins/volume/exec/nodeagent~uds
```

Comment:
This, alongside most of the other settings (mtu aside) are the defaults. Do we need to define all of them? Or can just rely on the defaults?

Reply:
From what I get, the configuration applied via the k0sctl and by extension the k0s terraform resource, is being merged with the default for any values missing. (You can verify it by getting the running k0s.conf from the k0s backup command). However I think that specifying a setting explicitly saves up down the line if a default value changes during an upgrade.

Reply:
Generally, upgrading without reading tons of changelogs isn't a good practice and default changes are red flags. But ok, hardcoding won't hurt either. But there is one default, MTU, where the default value, 0, does autodetection of MTU